src/eko/output/struct.py

```diff
@@ -0,0 +1,502 @@
+    result of the computation is already saved
+    - for a new object, it is the path at which any result of final or
+    intermediate computation is stored, as soon as it is produced
+
+    The computation can be stopped at any time, without the loss of any of the
+    intermediate results.
+
+    Attributes
+    ----------
+    path : pathlib.Path
+        path on disk, to which this object is linked (and for which it is
+        essentially an interface)
+    Q02 : float
+        inital scale
+    xgrid : interpolation.XGrid
+        momentum fraction internal grid
+    pids : np.ndarray
+        array of integers, corresponding to internal PIDs
+    configs : Configs
+        specific configuration to be used during the calculation of these
+        operators
+    rotations : Rotations
+        manipulation information, describing the current status of the EKO (e.g.
+        `inputgrid` and `targetgrid`)
+    debug : Debug
+        debug configurations
+    version : str
+        library version used to create the corresponding file
+    data_version : str
+        specs version, to which the file adheres
+
+    """
+
+    # operators cache, contains the Q2 grid information
+    _operators: Dict[float, Optional[Operator]]
+    # public attributes
+    # -----------------
+    # mandatory, identifying features
+    path: pathlib.Path
+    Q02: float
+    xgrid: interpolation.XGrid
+    pids: np.ndarray
+    # collections
+    configs: Configs
+    rotations: Rotations
+    debug: Debug
+    # tagging information
+    version: str = vmod.__version__
+    data_version: str = vmod.__data_version__
+
+    def __post_init__(self):
+        if self.path.suffix != ".tar":
+            raise ValueError("Not a valid path for an EKO")
+        if not tarfile.is_tarfile(self.path):
+            raise ValueError("EKO: the corresponding file is not a valid tar archive")
+
+    def __getitem__(self, q2: float):
+        # TODO: autoload
+        op = self._operators[q2]
+        if op is not None:
+            return op
+
+        start = f"operators/{q2:8.2f}"
+
+        with tarfile.open(self.path) as tar:
+            names = list(filter(lambda n: n.startswith(start), tar.getnames()))
+
+            if len(names) == 0:
+                raise ValueError(f"Q2 value '{q2}' not available in '{self.path}'")
+            if len(names) > 1:
+                raise ValueError(
+                    f"Q2 value '{q2}' occurs multiple times in '{self.path}'"
+                )
+
+            name = names[0]
+            compressed = name.endswith(".lz4")
+            stream = tar.extractfile(name)
+
+            if stream is None:
+                raise ValueError
+
+            op = Operator.load(stream, compressed=compressed)
+
+        self._operators[q2] = op
+        return op
+
+    def __setitem__(self, q2: float, op: Optional[Operator], compress: bool = True):
+        if op is not None:
```

Comment:
Why would it be `None`?

Reply:
I guess in case you want to erase it? but that's just stupid, I'll drop